src/main/java/beam/physsim/cch/RoutingFrameworkTravelTimeCalculator.scala

```diff
@@ -0,0 +1,251 @@
+package beam.physsim.cch
+
+import java.util
+import java.util.concurrent.Executors
+
+import beam.agentsim.events.PathTraversalEvent
+import beam.sim.BeamServices
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.{Coordinate, Envelope}
+import com.vividsolutions.jts.index.kdtree.{KdNode, KdTree}
+import org.apache.commons.lang.time.StopWatch
+import org.matsim.api.core.v01.network.Link
+import org.matsim.core.controler.events.IterationEndsEvent
+
+import scala.collection.JavaConverters._
+import scala.collection.convert.ImplicitConversionsToScala._
+import scala.collection.mutable
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+class RoutingFrameworkTravelTimeCalculator(
+  private val beamServices: BeamServices,
+  private val osmInfoHolder: OsmInfoHolder,
+  private val routingFrameworkWrapper: RoutingFrameworkWrapper,
+) extends LazyLogging {
+
+  private implicit val executionContext: ExecutionContext = ExecutionContext.fromExecutorService(
+    Executors.newFixedThreadPool(
+      Math.max(Runtime.getRuntime.availableProcessors() / 4, 1),
```

Comment:
remove division

Reply:
add documentation if this might cause issue

Reply:
done